### ISSUE TITLE
Default language unreachable when preferred browser language is different

### ISSUE THREAD

**Issue by pamtbaau** [getgrav/grav#2948]
With a certain language setting, the default language (the first in the list of supported languages) cannot be reached and visitor is redirected to the preferred language set by the browser.

For example:

A site having the following language settings:
```
languages:
  supported:
    - de
    - en
    - es
  include_default_lang: false
  http_accept_language: true
```

When the preferred language of the browser is set to Spanish the following will happen:
- localhost/**de**/typography -> localhost/**es**/typograpy => German is unreachable
- localhost/**en**/typography -> localhost/**en**/typograpy
- localhost/**es**/typography -> localhost/**es**/typograpy

When either of `include_default_lang: false` or `http_accept_language: true` is changed, the issue is solved, but then you loose the required behaviour.

Tested on both v1.6.26 and v1.7.0-rc.12

**Comment by mahagr** [comment]
Do you mean that Grav uses the browser language even if the URL specifies the wanted language?

**Comment by pamtbaau** [comment]
Yes, if the user specifies the language in the URL the user will be redirected to the language of the browser.
- With no language specified in the URL -> redirection to browser language
localhost/typography -> localhost/**es**/typograpy 
- With language specified in URL, user will also be redirected -> first language in supported languages becomes unreachable
localhost/**de**/typography -> localhost/**es**/typograpy

I've spent quite some time debugging Language.php, but the logic appears ungraspable to me... :-)

**Comment by mahagr** [comment]
I just checked the code and Grav prefers the language prefix over content negotiation. It also stores language into session (if you're using language prefix) and uses that over the language set in the browser.

**Comment by pamtbaau** [comment]
Did you actually test it?

Steps to reproduce:
- Create new Grav 1.6.26 site using latest download
- Add the following to /user/config/system.yaml
  ```
  languages:
    supported:
      - de
      - en
      - es
    include_default_lang: false
    http_accept_language: true
  ```
- Create following page structure:
  ```
  user/pages
  └── 01.home
      ├── default.de.md
      ├── default.en.md
      └── default.es.md
  ```
- Set preferred language of browser to any language other than German
- Point browser at: 
  - .../en&nbsp;&nbsp;->&nbsp;&nbsp;.../en
  - .../es&nbsp;&nbsp;->&nbsp;&nbsp;.../es
  - .../de&nbsp;&nbsp;->&nbsp;&nbsp;.../en  &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;=> German unreachable
  - .../&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;->&nbsp;&nbsp;.../en
 
  Note: Above results are with browser language set to English

My env:
- Windows 10 v2004
- WSL 1.0 running Ubuntu 20.04
- All extension in browser switched off

**Comment by rhukster** [comment]
What browser are you using, and how are you changing your preferred language in your browser?

**Comment by pamtbaau** [comment]
I'm using Chrome Version 83.0.4103.116
I've set the language via: chrome://settings/languages

**Comment by onetrev** [comment]
I'm on Grav 1.7.0-rc.13 but I've noticed something similar (maybe the same thing??). Suddenly all my URLs are redirecting to `/en/` in front of them. For example my hard coded link to `/about/` suddenly started 404-ing because it is auto-redirected now to '/en/about/`. 

Disabling "Include default language" option solved the issue, but it seems like this shouldn't just start happen all of a sudden right? I'm using Firefox 78.0.1.

**Comment by rhukster** [comment]
Can you try the latest in GitHub 1.7 branch. We reverted some changes for this reason.

**Comment by pamtbaau** [comment]
@rhukster Who do you ask?

**Comment by rhukster** [comment]
That was for @onetrev

**Comment by onetrev** [comment]
Will do @rhukster, thanks! So far it looks good but I'll have to test more when I can.

**Comment by pamtbaau** [comment]
@rhukster @mahagr Is there any progress on the original issue posted?

**Comment by rhukster** [comment]
We had to roll this fix back as it broke too many other things.  We'll have to revisit it later.

**Comment by goraht** [comment]
I am facing the same problem. Are there any chances it will be fixed any time soon?
